### PR TITLE
Elaborator Support

### LINKED ISSUE #28: Integrate parsing & typechecking into LSP
Once parsing and typechecking are complete, would be great to integrate them into LSP.

### PR DESCRIPTION
**Pull request by inttik**
Support elaborating in compilation and LSP.

### CODE REVIEW

File: snapshot_lsp_sample_elaboration.snap
Comment (inttik) [inline snapshot_lsp_sample_elaboration.snap:1]: That is a file, that should be deleted from PR, but there is some unexpected behaivour from elaborating, which fails some lsp tests. Sample can be found [here](https://github.com/DependoBuf/dependobuf/blob/inttik/support-elaborator/dbuf-lsp/tests/sample.dbuf).

File: snapshot_lsp_sample_elaboration.snap [hunk 1]
```diff
@@ -0,0 +1,1959 @@
...
+                        },
+                    ),
+                ],
+                constructor_names: OfMessage(
+                    InternedString {
+                        inner: "M2",
+                    },
+                ),
+            },
+        ),
```
Comment (inttik) [inline snapshot_lsp_sample_elaboration.snap]: That is type definition of M2 message (source [here](https://github.com/DependoBuf/dependobuf/blob/dc889122776c3181e0efc7811fa6acacb362004e/dbuf-lsp/tests/sample.dbuf#L7-L12)). Everything alright, expect dependencies has strange names: (with extra _dep). That feels wrong.

File: snapshot_lsp_sample_elaboration.snap [hunk 2]
```diff
@@ -0,0 +1,1959 @@
...
+                                    name: InternedString {
+                                        inner: "String",
+                                    },
+                                    dependencies: [],
+                                },
+                            },
+                        ],
+                    },
+                ),
+            ],
```
Comment (inttik) [inline snapshot_lsp_sample_elaboration.snap]: That (and further) is constructor M2. Here implicits (in current context - dependencies) have expected names without _dep.

**Comment by inttik** [comment]
Should be merged after #91

File: snapshot_lsp_sample_elaboration.snap
Comment (TurtlePU) [reply snapshot_lsp_sample_elaboration.snap:1]: Is it fixed in #91?

File: dbuf-lsp/tests/action/rename_tests.rs
Comment (TurtlePU) [inline dbuf-lsp/tests/action/rename_tests.rs:1]: Can some of this be checked via snapshot testing?

**Review by TurtlePU — DISMISSED** [review]

File: dbuf-lsp/tests/action/rename_tests.rs
Comment (inttik) [reply dbuf-lsp/tests/action/rename_tests.rs:1]: It was checked with variation of snapshot testing. Current approach is better since it contains small files, so bugs are easier to find if any, and it is easier to modify, so no need to calculate locations by hand to check if answers are correct. But the file itself is quite wordly. I think no need to be so pedantic in others modules.

File: snapshot_lsp_sample_elaboration.snap
Comment (inttik) [reply snapshot_lsp_sample_elaboration.snap:1]: Yes, It is fixed. There is should be a problem only with one test case, but since it is elaboration problem, not lsp one, we can keep the test as should panic or paraphrase it so no corner case of elaboration presented.

**Review by TurtlePU — APPROVED** [review]
